synapse/handlers/room.py

```diff
@@ -732,6 +738,15 @@ async def create_room(
         invite_3pid_list = config.get("invite_3pid", [])
         invite_list = config.get("invite", [])
 
+        # validate each entry for correctness
+        for invite_3pid in invite_3pid_list:
+            if not self._has_all_3pid_keys(invite_3pid):
+                raise SynapseError(
+                    400,
```

Comment:
generally we try to use `http.HTTPStatus` constants here (you may need to add an import):

```suggestion
                    HTTPStatus.BAD_REQUEST,
```

likewise in the other place you do this.